### PR TITLE
feat: return full nextQuery object in paginated calls

### PR DESCRIPTION
**Pull request by AVaksman**
- [x] Ensure the tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)

Combine a `nextPageRequest` from server with user provided custom options, this way user can just pass it to the next call as it is documented in the examples

https://github.com/googleapis/nodejs-spanner/blob/d835ba21489885f1fc092011d955c9ca5ef162ca/src/instance.ts#L1159-L1171

### CODE REVIEW

File: src/database.ts
Comment (stephenplusplus) [inline src/database.ts:1339]: `nextPageRequest`

Comment (stephenplusplus) [inline src/database.ts:1339]: What is the format/properties of `nextPageRequest`?

File: src/index.ts
Comment (stephenplusplus) [inline src/index.ts:537]: `nextPageRequest`*

File: src/index.ts
Comment (stephenplusplus) [inline src/index.ts:743]: `nextPageRequest`*

File: src/instance.ts
Comment (stephenplusplus) [inline src/instance.ts:354]: `nextPageRequest`*

File: src/instance.ts
Comment (stephenplusplus) [inline src/instance.ts:551]: `nextPageRequest`*

File: src/instance.ts
Comment (stephenplusplus) [inline src/instance.ts:675]: `nextPageRequest`*

File: src/instance.ts
Comment (stephenplusplus) [inline src/instance.ts:1239]: `nextPageRequest`*

**Review by stephenplusplus — CHANGES_REQUESTED** [review]

File: src/database.ts
Comment (AVaksman) [reply src/database.ts]: Thanks,
Fixed here and everywhere below.

Comment (AVaksman) [reply src/database.ts]: `nextPageRequest` comes in as `proto.~.IList<resource>Request` which is usually
```ts
parent/database;
pageSize;
pageToken;
//some request objects also have 
filter
```

**Review by stephenplusplus — APPROVED** [review]

**Comment by stephenplusplus** [comment]
@AVaksman should we merge, or do we need to wait for another review?

**Comment by stephenplusplus** [comment]
It seemed mergeable. Hopefully that wasn't a mistake!

**Pull request merged**
